### ISSUE TITLE
Require hook doesn't load plugins properly

### ISSUE THREAD

**Issue by lukescott** [babel/babel#1329]
It seems that when the require hook is used babel tries to load plugins for every plugin loaded.

Code: https://dl.dropboxusercontent.com/u/6619505/repo.zip

Run:

```bash
npm install
./node_modules/.bin/mocha
```

I get:

```
/Users/lukescott/Desktop/repo/node_modules/babel-core/lib/babel/transformation/file/index.js:285
      throw new ReferenceError("The key for plugin " + JSON.stringify(name) +
            ^
ReferenceError: The key for plugin "class-name" of addClassName collides with an existing plugin
```

If I convert the code in ./node_modules/babe-plugin-class-name to ES6 (w/ import and export) I get:

```
(function (exports, require, module, __filename, __dirname) { import {Transfor
                                                              ^^^^^^
Unexpected reserved word
```

**Comment by lukescott** [comment]
A little more info on this. I've been having trouble debugging this because when I changed something in babel to make it work, then changed it back it would work again. Figured out that the result of the plugin is cached in ~/.babel.json.

Still trying to figure out the cause.

**Comment by lukescott** [comment]
Ug. Found the problem. .babelrc is being loaded twice and plugins end up in the list more than once. I'll prepare a PR.

**Comment by sebmck** [comment]
Oh yeah, that's right. The require hook will resolve the config before transpiling so it can honour `ignore` and `only`. It'd be loaded a second time on transform internally since an absolute path is passed.

**Comment by sebmck** [comment]
Add `breakConfig = true` to the options object that is passed to `babel.transform` in `babel/register`.

**Comment by lukescott** [comment]
@sebmck Do you want me to send you a PR? I think I have a way of fixing this.

**Comment by sebmck** [comment]
@lukescott Sure :)

**Comment by lukescott** [comment]
@sebmck I've prepared PR #1335 that fixes this issue and possibly any future problems with `.babelrc`.

**Comment by sebmck** [comment]
Released as of 5.1.12, thanks!
